example/particles/particles.cpp

```diff
@@ -595,8 +627,10 @@ TaskCollection ParticleDriver::MakeFinalizationTaskCollection() const {
 
     auto destroy_some_particles = tl.AddTask(none, DestroySomeParticles, pmb.get());
 
-    auto deposit_particles =
-        tl.AddTask(destroy_some_particles, DepositParticles, pmb.get());
+    auto sort_particles =
+        tl.AddTask(destroy_some_particles, SortParticlesIfNecessary, pmb.get());
```

Comment:
What defines "is necessary" here? I'm not sure I follow the intention of this naming (in the context of the example).

Reply:
Sorry, should've responded before, this sort is necessary if we're using the pre-sorted per-cell lists for the non-atomic branch of the subsequent particle deposition call (which assumes those sorted lists exist). `SortParticlesIfNecessaryForDeposition`?

Reply:
Changed to `SortParticlesIfUsingPerCellDeposition`